packages/odf/components/kms-config/azure-config.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { FieldLevelHelp } from '@odf/shared/generic/FieldLevelHelp';
+import { useDeepCompareMemoize } from '@odf/shared/hooks/deep-compare-memoize';
+import { FileUploadInput } from '@odf/shared/modals/FileUploadInput';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import * as _ from 'lodash-es';
+import { FormGroup } from '@patternfly/react-core';
+import { AzureConfig, ProviderNames } from '../../types';
+import { AzureKMSConnectionForm } from './azurekms-connection-form';
+import { KMSConfigureProps } from './providers';
+import { kmsConfigValidation } from './utils';
+
+export const AzureConfigure: React.FC<KMSConfigureProps> = ({
+  state,
+  dispatch,
+  className,
+}) => {
+  const { t } = useCustomTranslation();
+  const azureState = useDeepCompareMemoize(
+    state.kms.providerState,
+    true
+  ) as AzureConfig;
+  const azureStateClone: AzureConfig = React.useMemo(
+    () => _.cloneDeep(azureState),
+    [azureState]
+  );
+
+  const updateAzureState = React.useCallback(
+    (kmsConfig: AzureConfig) =>
+      dispatch({
+        type: 'securityAndNetwork/setKmsProviderState',
+        payload: kmsConfig,
+      }),
+    [dispatch]
+  );
+
+  React.useEffect(() => {
+    const hasHandled: boolean = kmsConfigValidation(
+      azureState,
+      ProviderNames.AZURE
+    );
+    if (azureState.hasHandled !== hasHandled) {
+      updateAzureState({ ...azureState, hasHandled });
+    }
+  }, [updateAzureState, azureState]);
+
+  return (
+    <>
+      {/* TODO: Move AzureKMSConnectionForm here! */}
+      <AzureKMSConnectionForm
+        className={className}
+        kmsState={azureStateClone}
+        updateKmsState={updateAzureState}
+      />
+      <FormGroup>
+        <FileUploadInput
+          key={`azure-clientCert`}
+          id={'clientCert'}
+          inputName={t('Certificate')}
+          labelIcon={
+            <FieldLevelHelp>
+              {t(
+                "Certificates serve as secrets to authenticate the application's identity when requesting a token. They are also known as public keys."
+              )}
+            </FieldLevelHelp>
+          }
+          helperText={t(
+            'File must include a client certificate and a private key.'
+          )}
+          className={className}
+          kmsState={azureStateClone}
```

Comment:
why are you not passing all the props that were originally needed by this component ??
```suggestion
          className={className}
          kmsState={azureState}
          kmsStateClone={azureStateClone}
```